### PR TITLE
Add edit & delete doggo features along with tests

### PR DESCRIPTION
**Pull request by asawo**
**What**
User will be able to edit/delete doggos from "My doggos" screen.

**Why** 
Because it's a fundamental & crucial feature of this service. Tests should be written to check reliability and for debugging purposes.

### CODE REVIEW

File: server/routes/functions.js [hunk 1]
```diff
@@ -12,19 +12,14 @@ const registerUser = (req, res) => {
...
+				return db.createAccount(formData)
 			}
 		})
+		.then(response => {
+			res.status(200).json({ Success: `${response.formData.username} created` })
+		})
 		.catch(error => {
-			res.status(500).json({ error: error.message }) // look into the right http status codes to return
-			console.log('error', error.message)
+			console.log('registerUser error: ', error.message)
```
Comment (VeysonD) [inline server/routes/functions.js]: How will you handle these errors?

Comment (asawo) [reply server/routes/functions.js]: Good question... I may have deleted this part on accident 😇

File: client/home.js [hunk 2]
```diff
@@ -40,7 +40,6 @@ const logOutBtn = document.querySelector('#logOut')
 
 logOutBtn.addEventListener('click', async e => {
 	let response = await fetch('/logout')
```
Comment (VeysonD) [inline client/home.js]: If response isn't going to be redefined somewhere else, then using `const` is preferred.

Same goes for `resJson` on the next tline

File: client/my-doggo.js [hunk 3]
```diff
@@ -48,4 +47,92 @@ async function loadDoggos() {
 	}
 }
 
+const logOutBtn = document.querySelector('#logOut')
+
+logOutBtn.addEventListener('click', async e => {
+	let response = await fetch('/logout')
```
Comment (VeysonD) [inline client/my-doggo.js]: `const` instead of `let`

File: client/my-doggo.js [hunk 4]
```diff
@@ -48,4 +47,92 @@ async function loadDoggos() {
...
+const deleteRequest = async (data = {}) => {
+	const response = await fetch('/users/delete-doggo', {
+		headers: {
+			'Content-Type': 'application/json'
+		},
+		method: 'DELETE',
+		body: JSON.stringify(data)
+	})
+
+	let status = response.status
```
Comment (VeysonD) [inline client/my-doggo.js]: `const` instead of `let` here

File: client/my-doggo.js [hunk 5]
```diff
@@ -48,4 +47,92 @@ async function loadDoggos() {
...
+
 loadDoggos()
+	.then(result => {
+		// Add event listener on every doggo card's edit button
+		const editDoggo = document.querySelectorAll('.editDog')
+
+		editDoggo.forEach(doggo =>
+			doggo.addEventListener('click', async e => {
+				const doggoName =
+					e.target.parentElement.parentElement.nextElementSibling
```
Comment (VeysonD) [inline client/my-doggo.js]: There has to be a better way to find this element

Maybe using an id/class and finding it that way using a selector?

File: client/my-doggo.js (hunk 4 above)
Comment (asawo) [reply client/my-doggo.js]: Done!

File: client/home.js (hunk 2 above)
Comment (asawo) [reply client/home.js]: Done

File: server/routes/functions.js (hunk 1 above)
Comment (asawo) [reply server/routes/functions.js]: Put it back in

File: client/my-doggo.js (hunk 5 above)
Comment (asawo) [reply client/my-doggo.js]: I was thinking the same, like I was thinking of just adding an id to the edit button with the dogName, but a dog name can have a space in it, which means I can't use it for class/id

**Pull request merged**